vignettes/symptom-surveys.Rmd

```diff
@@ -0,0 +1,672 @@
+      filter(complete.cases(.)),
+    outcome = "case",
+    predictors = c("case", "fb"),
+    trainer = quantile_reg(),
+    args_list = arx_args_list(
+      lags = lags,
+      ahead = .x,
+      nonneg = FALSE
+    )
+  )$predictions %>%
+    mutate(lead = .x) %>%
+    left_join(z_te %>% filter(target_date == (date + .x)), by = c("geo_value", "target_date"))) %>%
+    list_rbind() %>%
+    mutate(err2 = abs(inv_trans(.pred) - inv_trans(target_case))) %>%
+    select(geo_value, forecast_date, err2, lead)
+
+
+  # Cases and Google model
+  out_df3 <- map(leads, ~ arx_forecaster(
+    z %>%
+      filter(between(time_value, date - .x - max(lags) - n, date)) %>%
+      select(time_value, geo_value, case, goog) %>%
+      filter(complete.cases(.)),
+    outcome = "case",
+    predictors = c("case", "goog"),
+    trainer = quantile_reg(),
+    args_list = arx_args_list(
+      lags = lags,
+      ahead = .x,
+      nonneg = FALSE
+    )
+  )$predictions %>%
+    mutate(lead = .x) %>%
+    left_join(z_te %>% filter(target_date == (date + .x)), by = c("geo_value", "target_date"))) %>%
+    list_rbind() %>%
+    mutate(err3 = abs(inv_trans(.pred) - inv_trans(target_case))) %>%
+    select(geo_value, forecast_date, err3, lead)
+
+  # Cases, Facebook and Google model
+  out_df4 <- map(leads, ~ arx_forecaster(
+    z %>%
+      filter(between(time_value, date - .x - max(lags) - n, date)) %>%
+      select(time_value, geo_value, case, fb, goog) %>%
+      filter(complete.cases(.)),
+    outcome = "case",
+    predictors = c("case", "goog"),
+    trainer = quantile_reg(),
+    args_list = arx_args_list(
+      lags = lags,
+      ahead = .x,
+      nonneg = FALSE
+    )
+  )$predictions %>%
+    mutate(lead = .x) %>%
+    left_join(z_te %>% filter(target_date == (date + .x)), by = c("geo_value", "target_date"))) %>%
+    list_rbind() %>%
+    mutate(err4 = abs(inv_trans(.pred) - inv_trans(target_case))) %>%
+    select(geo_value, forecast_date, err4, lead)
+
+  # Left join of the results for all models
+  out_list[[k]] <- left_join(left_join(left_join(left_join(out_df0, out_df1), out_df2), out_df3), out_df4)
+}
+# Outside of loop bind rows of list
+out_df <- do.call(rbind, out_list)
+```
+
+## Results: All Four Models
+
+Since there are only two common forecast dates available for the four models for the 14-day-ahead forecasts (May 13 and May 14, 2020), we skip studying the 14-day-ahead forecast results in this four-way model discussion. 
+
+Below we compute the median scaled errors for each of the four models over the 9-day test period. We can see that adding either or both of the survey signals improves on the median scaled error of the model that uses cases only, with the biggest gain achieved by the "Cases + Google" model. We can also see that the median scaled errors are all close to 1 (with all but that from the "Cases + Google" and "Cases + Facebook + Google" models exceeding 1), which speaks to the difficulty of the forecasting problem.
```

Comment:
Why only 9 days? Can we make this larger now?

Reply:
The reason for this shortened period: we paused running the Google survey on May 14 so its data ends there, but as we explained in our last post, we [plan to bring it back](https://delphi.cmu.edu/blog/2020/09/18/covid-19-symptom-surveys-through-google/#google-survey-redux) later that fall. So I was unable to find a sustained period of overlap for the two surveys over the start and end days that we considered.